### ISSUE TITLE
IdentityToken.IsExpired is always true

### ISSUE THREAD

**Issue by hapthien** [openstacknetsdk/openstack.net#630]
Hi guys,

Thank you for reviewing my reported issue.
I have a problem with checking IdentityToken.IsExpired flag. It is always True if the date time format of workstation is not the same with the expires value returned by server.
I got problem with
 - OVH Cloud
 - Server in French culture

https://github.com/openstacknetsdk/openstack.net/blob/develop/src/corelib/Core/Domain/IdentityToken.cs

In this file : https://github.com/openstacknetsdk/openstack.net/blob/develop/src/corelib/Core/Domain/IdentityToken.cs
-Should the expires value be in DateTime?
-and when checking the IsExpired, should consider about the Culture when "TryParse"?

Thank you,
hapthien

**Comment by carolynvs** [comment]
Thanks for reporting this problem. I think you may be right that it should instead being using `ParseExact` and specifying the formatstring, instead of relying upon the current culture but I'll need to verify that.

**Comment by carolynvs** [comment]
I'm having trouble reproducing the problem. Would you please supply a bit more information:

* The string value of `expires` returned from the identity api.
* The current culture: `CultureInfo.CurrentCulture.Name`.
* Does the following code work? If the parsing fails, it assumes that it is expired. This should help us figure out if the problem is parsing or date comparison.

```
var expires = "the string value of expires, e.g. 2016-09-08T17:47:44.930Z";
DateTimeOffset expiration;
var success = DateTimeOffset.TryParse(expires, out expiration);
Debug.Assert(success);
Debug.Assert(DateTimeOffset.UtcNow < expiration);
```

**Comment by hapthien** [comment]
Sorry for the late reply.
The following is information you ask when I had issue with IsExpired always True
 - Expires value from OVH Public Cloud (the bases on Openstack):  "09/20/2016 03:21:27" (without quotes)
 - CultureInfo.CurrentCulture.Name value: "fr-BE" (without quotes)
 - var success = DateTimeOffset.TryParse(expires, out expiration); --> success = False
 - DateTimeOffset.UtcNow < expiration --> False

with the same code (I mean code for checking IsExpired and logging the above data), if I change my DateTime format as this image. It works well
![image](https://cloud.githubusercontent.com/assets/20165122/18621948/dca6b134-7e53-11e6-9beb-0a44d4b20137.png)

Thank you for your time to verify my problem.

**Comment by carolynvs** [comment]
Oh! Wow, I did not expect that their cloud would send a non-iso datetime for expires. That's why I wasn't able to reproduce, since vanilla openstack and rackspace API's both the ISO 8601 format, e.g. `2016-09-08T17:47:44.930Z`.

I'll see what I can do to support that...

**Comment by hapthien** [comment]
@carolynvs : I sent an email to your email address ( carolyn.vanslyck@rackspace.com) with sensitive information.
Thank you for your time.

**Comment by carolynvs** [comment]
Thanks for the info! I am able to reproduce and am working on a fix today.

**Comment by carolynvs** [comment]
Wow, this has been a bug for quite a while. Here's a ⭐ for finding it and helping get it fixed!

You can get the fix in [v1.7.6 on nuget](https://github.com/openstacknetsdk/openstack.net/releases/tag/v1.7.6).

**Comment by hapthien** [comment]
It works as expected now.
Thank you.
